management_api_app/api/routes/workspaces.py

```diff
@@ -55,6 +57,42 @@ async def create_workspace(workspace_create: WorkspaceInCreate, workspace_repo:
     return WorkspaceIdInResponse(workspaceId=workspace.id)
 
 
+@router.post("/workspaces/{workspace_id}/workspace-services", status_code=status.HTTP_202_ACCEPTED,
+             response_model=WorkspaceServiceIdInResponse, name=strings.API_CREATE_WORKSPACE_SERVICE)
+async def create_workspace_service(workspace_create: WorkspaceServiceInCreate,
+                                   workspace_service_repo: WorkspaceServiceRepository = Depends(
+                                       get_repository(WorkspaceServiceRepository)),
+                                   user: User = Depends(get_current_user),
+                                   workspace: Workspace = Depends(get_workspace_by_workspace_id_from_path)
+                                   ) -> WorkspaceServiceIdInResponse:
+    access_service = get_access_service()
+    if access_service.get_workspace_role(user, workspace) != WorkspaceRole.Owner:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=strings.ACCESS_USER_IS_NOT_OWNER)
+
+    try:
+        workspace_service = workspace_service_repo.create_workspace_service_item(workspace_create, workspace.id)
+    except (ValidationError, ValueError) as e:
+        logging.error(f"Failed create workspace service model instance: {e}")
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
```

Comment:
ideally we don't want to return the error directly to the user (as it can contain implementation specific details that could be used for security breaches) - do we know that this error is safe to return? I can see that it is done the same way in the workspaces api so depending on the answer, please change in both places.

Reply:
Those are ValidationError and ValueError telling the user about missing parameters and parameter value errors...  I think it is ok to return them, they inform the user about missing/invalid information that the template requires, and I think it is safe, because the user can call the template api by name and see what are the required parameters that are set there.

Reply:
cool with me